test/core/pin.js

```diff
@@ -275,6 +285,78 @@ describe('pin', function () {
             ])
           )
       })
+
+      it('direct for CID', function () {
+        return pin.ls(pins.mercuryDir, { type: 'direct' })
+          .then(out =>
+            expect(out).to.have.deep.members([
+              { type: 'direct',
+                hash: pins.mercuryDir }
+            ])
+          )
+      })
+
+      it('direct for path', function () {
+        return pin.ls(`/ipfs/${pins.root}/mercury/`, { type: 'direct' })
+          .then(out =>
+            expect(out).to.have.deep.members([
+              { type: 'direct',
+                hash: pins.mercuryDir }
+            ])
+          )
+      })
+
+      it('direct for path (no match)', function (done) {
+        pin.ls(`/ipfs/${pins.root}/mercury/wiki.md`, { type: 'direct' }, (err, pinset) => {
+          expect(err).to.exist()
```

Comment:
To make it easier to maintain the error message is not hardcoded here, but checked in interop tests: 
- https://github.com/ipfs/interface-js-ipfs-core/pull/375/files#diff-6a00eb1307f801bb37999815584589bdR190